### ISSUE TITLE
Time format not matching

### ISSUE THREAD

**Issue by nclm** [CopOnTheRun/FB-Message-Parser#5]
Hi, thanks for making this!

I get this error when parsing my archive:

```
ValueError: time data 'Friday, 19 June 2015 at 19:17 UTC+02' does not match format '%A, %B %d, %Y at %I:%M%p %Z'
```

I guess there can be format inconsistencies in an archive, and the tool doesn’t expect that.

**Comment by CopOnTheRun** [comment]
Yeah, I believe that converting the timestamp from the archived messages into a python `datetime` object is country specific. One of the previous contributors from the UK had to alter the source code for that portion as well.

**Comment by nclm** [comment]
Ok, thanks for your answer. Would this work too just by changing the preferred language on Facebook then exporting again? I’ll try.

**Comment by jsharkey13** [comment]
If you change [line 12](https://github.com/CopOnTheRun/FB-Message-Parser/blob/master/fb_parser.py#L12) in `fb_parser.py` to be `dtFormat = '%A, %d %B %Y at %H:%M %Z+02'` then it will parse that timestamp you copied above ok.

But if summer time becomes an issue (i.e. there are lines with UTC+01 or UTC+03) then that won't work. Changing [line 25](https://github.com/CopOnTheRun/FB-Message-Parser/blob/master/fb_parser.py#L25) to include `.replace("+01", "").replace("+02", "").replace("+03", "")` will get rid of the summer time part and allow it to parse ok. You'll get times which are an hour out from what you might expect though . . .

**Comment by andebor** [comment]
I have summertime and multiple timezones in my log, so I just changed [line 25](https://github.com/CopOnTheRun/FB-Message-Parser/blob/master/fb_parser.py#L25) to 
```
dt.strptime(re.sub(r'\+\d\d', '', y.find(class_='meta').string), dtFormat),
```
This will remove any `+XX` or `-XX`
and remember to add `import re`
You can compile the regular expression before the loop to speed things up a bit, but my dataset wasn't that big.

**Comment by jsharkey13** [comment]
Ok, so I finally got round to implementing the [correct fix on my fork](https://github.com/jsharkey13/facebook_message_parser/blob/master/fb_parser.py#L148) of this repo; the simplest way is to use `import dateutil` and use `dateutil.parser.parse`.

For this repo's code, this amounts to changing [fb_parser.py L25](https://github.com/CopOnTheRun/FB-Message-Parser/blob/master/fb_parser.py#L25) to be:
`dateutil.parser.parse(y.find(class_='meta').string).replace(tzinfo=None),`
and importing `dateutil`, I think (though I haven't tested it)!

**Comment by nclm** [comment]
@jsharkey13 Thank you a lot, with your fork I finally was able to parse my messages! (and made stats from them)
